adr/ADR-40.md

```diff
@@ -52,209 +61,426 @@ This method is available in all NATS Server versions.
 
 1. Clients initiate a network connection to the Server.
 2. Server responds with [INFO][INFO] json.
-3. If Server [INFO][INFO] contains `tls_required` set to `true`, or the client has a tls requirement set to `true`, the client performs a TLS upgrade.
+3. If Server [INFO][INFO] contains `tls_required` set to `true`, or the client
+   has a tls requirement set to `true`, the client performs a TLS upgrade.
```

Comment:
Is this the behavior going forward, because it's not the current behavior. The current behavior is that the client and server are suppose to match.

Reply:
The server supports the ability of offering non-tis connections.
If it does, and you have a tls configuration on the client, it assumes, you want tis.
If it doesn't it also asumes that.
But if you can specify something like `null` it implies you don't want tls for the connection (they are opting out).

Reply:
when the server supports tls and the client uses an URL like `nats://connect.ngs.global` then it would auto upgrade to use TLS, clients ought to be able to handle that